### ISSUE TITLE
Disabling --ha and --checkpoint on command line

### ISSUE THREAD

**Issue by zdenekc** [mesosphere/marathon#2272]
I've been experimenting with marathon, trying to run it in single node configuration without high availability and I have run into issue with turning `--ha` option off on command line. I have tried to issue 'standard' negative prefixes like `--no-ha` or `--noha`, but none of them worked.

I have no experience with scala, but I had a look into code and it seems to me that there is no way to turn `--ha` (or `--checkpoint`) options from command line.

The `--ha` option is defined as Plain Scallop option, with default set to True:

```scala
  lazy val highlyAvailable = opt[Boolean]("ha",
    descr = "Runs Marathon in HA mode with leader election. " +
      "Allows starting an arbitrary number of other Marathons but all need " +
      "to be started in HA mode. This mode requires a running ZooKeeper",
    noshort = true, default = Some(true))
```

The Scallop documentation describes Plain flag options behaviour like this:

> Special case of plain options are the flag options. They take no arguments, and evaluate to true if option was present on the command line or to false if it was not.

Which leads me to conclusion that if `--ha` is present, it is true. When it is not present, default is used and it is true. Plain options do not support `prefix` for negation like Toggle option, so there is no default prefix that could be used to negate it.

As I said, I have no experience with Scala language at all, so I might be mistaken. Should I be correct, then I think using Toggle option with `prefix='no-' should fix it.

**Comment by kolloch** [comment]
I came to the same conclusion. So I introduced a manual "disable" prefixed option for `--revive_offers_for_new_apps`. See [`ReviveOffersConfig.scala`](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/flow/ReviveOffersConfig.scala#L8-L20):

```scala
  lazy val disableReviveOffersForNewApps = opt[Boolean]("disable_revive_offers_for_new_apps",
    descr = "Disable reviveOffers for new or changed apps. (Default: use reviveOffers) ",
    default = Some(false))

  lazy val reviveOffersForNewApps = opt[Boolean]("revive_offers_for_new_apps",
    descr = "Whether to call reviveOffers for new or changed apps. " +
      "(Default: use reviveOffers, disable with --disable_revive_offers_for_new_apps) ",
    hidden = true,
    default = Some(true))
```

**Comment by kolloch** [comment]
Out of curiosity: Why would you want to disable the "HA" mode?

**Comment by zdenekc** [comment]
As I said, I have no experience with Scalla, but from Scallop documentation it seems like [Toggle option](https://github.com/scallop/scallop/wiki/Option-types-and-definitions) could be used to do that. But I've never tried it, so I may be missing something. Nevertheless, thanks for pointing those new options to me. I'll have a go on them

As for disabling ha/checkpoint flags. We have single node Mesos/Marathon cluster for our developers so that they can run and deploy their components the same way as it is done in production. With single node, we have run into some edge cases that does not work very well for us.

For example, when rebooting that single node and after Marathon starts, it takes 10-15 minutes before it notices all of its tasks are dead. Which is inconvenience.

My investigation lead me to Mesos slave recovery periods and recovery type, which allowed me some reduction but not enough. So I tried to disable checkpointing, assuming that without state to recover, slave would start faster (and we don't mind container restarts on development machine).

Turned out it can't be done :-).
